src/components/ranking-chart/ranking-chart-component.jsx

```diff
@@ -68,6 +68,22 @@ function RankingChart({
     }
   };
 
+  useEffect(() => {
```

Comment:
This seems to only be working when we sort SPI and not species, human modification or protection. Not sure why

Reply:
Also, this solution is very similar to the selectedLandMarineOption useEffect above. Do you think we could extract some function to avoid repeating code? Also, maybe tableRef.current would need to be a dependency

Reply:
I have add a `selectedCountryIndex` state to calculate dynamicaly the index of the country (it was only working with SPI because it was getting the id of the array and not the position in the sorted array)  but it doesn't work yet because it have a little delay.

Reply:
fixed